### PR TITLE
[PropertyInfo] Fix dock block lookup fallback loop

### PR DESCRIPTION
**Pull request by DerManoMann**
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

`getDocBlock()` in `PhpDocExtractor` implements a fallback loop when trying to lookup a dock block for a property (property, accessor method, mutator method). This relies on the individual lookups to return `null`.

Unfortunately, phpDocumentor will throw an `InvalidArgumentException` exception in case there is no dock block.

Currently the try/catch is wrapping the whole loop which means if the first lookup fails the loop is ended instead of moving on to the next one.

This PR moves the try/catch to each individual call of `$this->docBlockFactory->create()`.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Thanks for the PR and the clear description. Could you add a test case please?

**Comment by DerManoMann** [comment]
Ah, looks like i rebased on master, not 4.1 ... is there an easy fix for this or is it better to create a new PR, I wonder?

File: src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php [hunk 1]
```diff
@@ -194,7 +190,13 @@ private function getDocBlockFromProperty(string $class, string $property): ?DocB
...
         }
 
-        return $this->docBlockFactory->create($reflectionProperty, $this->contextFactory->createFromReflector($reflectionProperty->getDeclaringClass()));
+        try {
+            return $this->docBlockFactory->create($reflectionProperty, $this->contextFactory->createFromReflector($reflectionProperty->getDeclaringClass()));
+        } catch (\InvalidArgumentException $e) {
+            // No valid doc block
+        }
+
+        return null;
```
Comment (dunglas) [inline src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php]: You can return in the catch (same below)

Comment (DerManoMann) [reply src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php]: Was wondering ... I guess that a personal preference thing for me. I always feel it looks weird to not have a return at the end if the return type is not void.

**Review by lyrixx — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @DerManoMann.

**Comment by nicolas-grekas** [comment]
Merged on branch 3.4 by the way.

**Pull request merged**